static/src/javascripts-legacy/projects/facia/modules/ui/lazy-load-containers.js

```diff
@@ -16,30 +14,55 @@ define([
     var distanceBeforeLoad = detect.getViewport().height;
 
     return function () {
-        var $frontBottom = bonzo(qwery('.js-front-bottom')),
-            containers = qwery('.js-container--lazy-load'),
+        var containers = qwery('.js-container--lazy-load'),
             lazyLoad = throttle(function () {
                 if (containers.length === 0) {
                     mediator.off('window:throttledScroll', lazyLoad);
                 } else {
                     fastdom.read(function () {
-                        var scrollTop = window.pageYOffset,
-                            scrollBottom = scrollTop + bonzo.viewport().height,
-                            bottomOffset = $frontBottom.offset().top,
-                            $container;
-
-                        if (scrollBottom > bottomOffset - distanceBeforeLoad) {
-                            $container = bonzo(containers.shift());
-
-                            fastdom.write(function () {
-                                $container.removeClass('fc-container--lazy-load');
-                            });
-                        }
+                        var cs = split(containers, withinRange);
+
+                        containers = cs[1];
+
+                        fastdom.write(function () {
+                            cs[0].forEach(displayContainer);
+                        });
                     });
                 }
             }, 500);
 
         mediator.on('window:throttledScroll', lazyLoad);
         lazyLoad();
     };
+
+    // split :: [Element] -> (Element -> Bool) -> ([Element], [Element])
+    // invariant: { inA all fn ^ outA all (complement fn) | (inA, outA) = split xs fn }
+    // Split an array of elements into two arrays, one where element comply, one where they don't
+    function split(arr, testFn) {
```

Comment:
I think if we can keep the context down to just syntax where possible, and not semantics, then our codebase and discussions will be more productive. For example, in your example above, I don't need to check your specific implementation of `reduce`, because I assume it's the same as every other language. You didn't (can't, in this case) use a `fold`, and that conveys meaning to me.

I'll leave the separate question of in-situ or module-scope function for @regiskuckaertz , certainly feels like a matter of taste.